resotocore/resotocore/model/graph_access.py

```diff
@@ -117,6 +123,8 @@ def add_from_json(self, js: Json) -> None:
                 js.get("search", None),
                 js.get("replace", False) is True,
             )
+        elif "from" in js and "to" in js and js.get("edge_type") == EdgeType.outer:
```

Comment:
I would opt for a different format. One that allows 
`(from|from_spec) -> (to|to_spec)`
and stores the final edge type.